### DISCUSSION TITLE
Simplebluez Read Data

### DISCUSSION THREAD

**Discussion by stephenbates101** [simpleble/simpleble#286] [Q&A]
Hello,
I have written a program which is based on the simplebluez example, it connects to the device I want and lists the services and characteristics correctly however when I try and read the data for the characteristic I want it only returns one byte not a byte array as I would expect. 
I am using ->read() would it be better to use ->value() , both functions appear to call functions in GattCharacteristic1.cpp which do not appear to specify how many bytes to read.
Can anyone help please?

**Comment by kdewald** [comment]
`value` returns whatever is the latest cached value of the characteristic value.
`read` will trigger an update from your peripheral device.

The conversion from a byte array to a single byte you can do on your own. I have faith in your skills.
